drgn_tools/multipath.py

```diff
@@ -0,0 +1,345 @@
+    }
+    output_format = output_mapping.get(ps_type)
+    if output_format:
+        if ps_type == "service-time":
+            load = pi.in_flight_size.counter / pi.relative_throughput
+            print(
+                output_format.format(
+                    space,
+                    pi.in_flight_size.counter.value_(),
+                    space,
+                    pi.relative_throughput.value_(),
+                    space,
+                    pi.repeat_count.value_(),
+                    space,
+                    load.value_(),
+                )
+            )
+        elif ps_type == "round-robin":
+            print(output_format.format(space, pi.repeat_count.value_()))
+        elif ps_type == "queue-length":
+            print(
+                output_format.format(
+                    space,
+                    pi.repeat_count.value_(),
+                    space,
+                    pi.qlen.counter.value_(),
+                )
+            )
+        elif ps_type == "io-affinity":
+            print(
+                output_format.format(
+                    space,
+                    pi.refcount.refs.counter.value_(),
+                    space,
+                    pi.failed.value_(),
+                )
+            )
+        else:
+            print(output_format)
+    else:
+        print("\n\nPath selection policy %s not supported.\n\n" % ps_type)
+
+
+def disk_size_and_permission(bdev: Object) -> None:
+    """
+    Print Disk write permission.
+    Print Disk Size in bytes.
+
+    :param bdev: `struct block device *`
+
+    :returns: None
+    """
+    if blkdev_ro(bdev):
+        print("permission = r")
+    else:
+        print("permission = rw")
+    print("Size = %u" % bdev.bd_inode.i_size)
+
+
+def show_pgpaths(priority_group: Object) -> None:
+    """
+    Show pgpath data.
+
+    :param priority_group: `struct priority_group *`
+
+    :returns: None
+    """
+    path_num = 1
+    ps_type = priority_group.ps.type.name.string_().decode()
+    format_header = "{:10}{:11}{:12}{:10}"
+    print(
+        format_header.format("Path_No.", "Disk_Name", "DevName", "State"),
+        end="",
+    )
+    switch_header(ps_type)
+    for pgpath in for_each_pgpath(priority_group):
+        bdev = pgpath.path.dev.bdev
+        disk_name = blkdev_name(bdev)
+        dev_name = pgpath.path.dev.name.string_().decode()
+        state = "active" if pgpath.is_active else "not active"
+        pi = cast("struct path_info *", pgpath.path.pscontext)
+        format_val = "{:<3}{:<7}{:<2}{:<9}{:<1}{:<8}{:<3}{:<13}"
+        print(
+            format_val.format(
+                space,
+                path_num,
+                space,
+                disk_name,
+                space,
+                dev_name,
+                space,
+                state,
+            ),
+            end="",
+        )
+        switch_output(ps_type, pi)
+        path_num += 1
+    disk_size_and_permission(bdev)
```

Comment:
The "size" and "wp" from "multipath -ll" should be for multipath device, you should be able to get it from "mapped_device.bdev"  which is a  "struct block_device *", you can use generic helper from block.py to get its size and permission. That way you should dump it before stepping into priority group dumpping.

Reply:
Removed it. 
Added a new function in block.py blkdev_size(bdev) to get size.
And using blkdev_ro(bdev) to read permission.